File: backend/app/schemas/geo.py
from typing import List, Literal, Union, Dict, Any, Optional
from pydantic import BaseModel, Field

class GeoJSONPolygon(BaseModel):
    type: Literal["Polygon"]
    coordinates: List[List[List[float]]] # [[[lng, lat], [lng, lat], ...]]

class GeoJSONMultiPolygon(BaseModel):
    type: Literal["MultiPolygon"]
    coordinates: List[List[List[List[float]]]]

GeoJSONGeometry = Union[GeoJSONPolygon, GeoJSONMultiPolygon]

class GeoJSONFeature(BaseModel):
    type: Literal["Feature"] = "Feature"
    id: Optional[str] = None
    geometry: Dict[str, Any]
    properties: Dict[str, Any] = Field(default_factory=dict)

class GeoJSONFeatureCollection(BaseModel):
    type: Literal["FeatureCollection"] = "FeatureCollection"
    features: List[GeoJSONFeature]

File: frontend/src/types/geo.ts
export interface GeoJSONPolygon {
  type: "Polygon";
  coordinates: number[][][];
}

export interface GeoJSONMultiPolygon {
  type: "MultiPolygon";
  coordinates: number[][][][];
}

export type GeoJSONGeometry = GeoJSONPolygon | GeoJSONMultiPolygon;

export interface GeoJSONFeature {
  type: "Feature";
  id?: string;
  geometry: GeoJSONGeometry;
  properties: {
    id: string;
    project_id: string;
    project_name: string;
    project_category: string;
    name: string;
    description: string;
    area_sq_km: number;
    area_ha: number;
    created_at: string;
    [key: string]: any;
  };
}

export interface GeoJSONFeatureCollection {
  type: "FeatureCollection";
  features: GeoJSONFeature[];
}
